From f6c494def47688f7180c59d4eb7944bf822640f0 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Tue, 19 Apr 2016 15:07:18 +0200
Subject: [PATCH 091/100] IPA: terminate properly if view name lookup fails

Since commit 5a5f1e1053415efaa99bb4d5bc7ce7ac0a95b757 the view name
lookup is the last step in the subdomain lookup request. In case of an
error the request should be finished and no previous step should be
called again.

Resolves https://fedorahosted.org/sssd/ticket/2993

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
(cherry picked from commit 57d8b4b9254442a568838cb60ea16068965f2df0)
(cherry picked from commit e5fbaf42f87725e42a40e542c06d5f4cdf4c52c2)
---
 src/providers/ipa/ipa_subdomains.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/providers/ipa/ipa_subdomains.c b/src/providers/ipa/ipa_subdomains.c
index f13847f12a7eae42b13a51e3fe1d09b60878633b..82d9468fa08c1aa9adff74be1d7609e3a2e48882 100644
--- a/src/providers/ipa/ipa_subdomains.c
+++ b/src/providers/ipa/ipa_subdomains.c
@@ -840,13 +840,7 @@ static void ipa_get_view_name_done(struct tevent_req *req)
         if (ret == EOPNOTSUPP || ret == EIO) {
             DEBUG(SSSDBG_TRACE_FUNC, "get_view_name request failed, looks " \
                                      "like server does not support views.\n");
-            ret = ipa_check_master(ctx);
-            if (ret == EAGAIN) {
-                return;
-            } else if (ret != EOK) {
-                goto done;
-            }
-
+            ret = EOK;
         } else {
             DEBUG(SSSDBG_OP_FAILURE, "get_view_name request failed.\n");
         }
-- 
2.7.4

